### ISSUE TITLE
`metrics-es-node-graphite.rb` creates some metrics un-parseable by graphite

### ISSUE THREAD

**Issue by TheKevJames** [sensu-plugins/sensu-plugins-elasticsearch#49]
I'm currently using this plugin to send metrics to Graphite; it's been mostly smooth sailing, but I've been getting a bunch of repeating warnings in my Carbon logs that I think can and should be cleaned up.

Carbon logs:

```
30/09/2016 21:04:10 :: [listener] invalid line (elasticsearch.es_example_net.indices.indexing.is_throttled false 1475269450) received from client 172.16.238.17:54114, ignoring
30/09/2016 21:04:10 :: [listener] invalid line (elasticsearch.es_example_net.fs.total.spins true 1475269450) received from client 172.16.238.17:54114, ignoring
```

Basically, Graphite does not support "true" or "false" for metrics -- it instead uses "1" and "0". Since these logs are generated from running `metrics-es-node-graphite.rb`, I would expect the output to be Graphite-compatible.

The offending issue is that the [elasticsearch metrics endpoint](https://github.com/sensu-plugins/sensu-plugins-elasticsearch/blob/master/bin/metrics-es-node-graphite.rb#L179) returns values such as `"spins": "true"`. To support Graphite output, we would need to encode this as integers, for example by changing [the output command](https://github.com/sensu-plugins/sensu-plugins-elasticsearch/blob/master/bin/metrics-es-node-graphite.rb#L294) to

```
output_value = case v
  when "true"
    1
  when "false"
    0
  else
    v
end
output([config[:scheme], k].join('.'), output_value, timestamp)
```

**Comment by majormoses** [comment]
Do you think you could submit a pr?

**Comment by Evesy** [comment]
Is this something that could be incorporated into https://github.com/sensu-plugins/sensu-plugin/blob/master/lib/sensu-plugin/metric/cli.rb to cover this issue across the board, or would it be better to just do on a per metric check basis when needed?

**Comment by majormoses** [comment]
I am not sure we want to blindly make that across the board, that being said if we added a helper function then it could be check specific and could maybe make it act something like this:
```
output([config[:scheme], k].join('.'), cast_bool_value(int), timestamp)
```

I think this gives us the best of both worlds, make peoples lives easier and give max flexibility.

**Comment by Evesy** [comment]
Sorry, were you thinking a helper function within the scope of this plugin, or one within the Sensu plugin graphite class?

**Comment by majormoses** [comment]
within the `sensu-plugin` gem this way it could be used across all plugins where it makes sense. I will make the example above a bit more clear.

**Comment by Evesy** [comment]
@majormoses Would you mind having a quick glance at https://github.com/Evesy/sensu-plugin/commit/7fd2b4c930cb77feb4d7d9a59858c315c3fb2615 to see if it's anything near what you were thinking?

**Comment by majormoses** [comment]
yup that is what I had in mind. Ping me when you create the PR and link it to this issue.

**Comment by majormoses** [comment]
So the upstream plugin has the library function now we just need to call it.

**Comment by majormoses** [comment]
Actually just checking is there anything else we need done here?

**Comment by TheKevJames** [comment]
@majormoses This issue is fixed for me with the merge of #85 into release 1.5.1. Not sure what the bool-to-int method had to do with this issue, in the end, but always nice to see functionality like that upstream. :)
